plugins/modules/organization.py

```diff
@@ -87,11 +171,50 @@ def main():
             name=dict(required=True),
             description=dict(),
             label=dict(),
+            upstream_type=dict(required=False, choices=['redhat_cdn', 'export_sync', 'network_sync']),
+            upstream_url=dict(required=False),
+            upstream_username=dict(required=False),
+            upstream_password=dict(required=False, no_log=True),
+            # upstream_ca_cert=dict(required=False, type='entity', resource_type='content_credentials', scope=['']),
+            upstream_ca_cert_id=dict(required=False),
+            upstream_organization=dict(required=False),
+            upstream_lifecycle_environment=dict(required=False),
+            upstream_content_view=dict(required=False),
         ),
     )
 
     with module.api_connection():
-        module.run()
+        handle_cdn_configuration = 'upstream_type' in module.foreman_params
+
+        organization = module.lookup_entity('entity')
+        new_organization = module.run()
+
+        if handle_cdn_configuration:
+            if organization:
```

Comment:
Doesn't this mean it only performs the "handle cdn config" part when updating an *existing* Org, and never when creating an org, thus requiring the playbook to run twice?
Also, this will probably fail in the `state=absent` case.

How about this?

```suggestion
        organization = module.run()

        if handle_cdn_configuration and not module.desired_absent:
```

Reply:
Hmm, now that I looked at other places… For some configurations, this requires the Org to already exist as it references objects *inside* the Org (Content Credentials).

I wonder if it's easier to make a standalone module for this? *scratches head*

Reply:
Oh, derp, yeah, it's not required when removing an org :-)

Also, yes, with your suggestion it should also work better when creating new orgs, the only exception is when creating a new org with ISS Network Sync, as that requires the org to contain a Content Credential